### PR TITLE
add ability to load `.dds`, `.tga`, and `.jpeg` texture formats

### PR DESCRIPTION
**Pull request by blunted2night**
controled by non-defaulted features of the same name

### CODE REVIEW

File: crates/bevy_render/src/texture/image_texture_loader.rs [hunk 1]
```diff
@@ -9,6 +9,32 @@ use bevy_utils::BoxedFuture;
...
+
+            )+ else { None }
+        }
+
+    };
+}
+
+define_file_extensions! {
+    "png" => Png, "dds" => Dds,
+    "tga" => Tga, "jpeg" => Jpeg,
```
Comment (rod-salazar) [inline crates/bevy_render/src/texture/image_texture_loader.rs:35]: What about "jpg" without the e?

Comment (blunted2night) [reply crates/bevy_render/src/texture/image_texture_loader.rs]: That would be the more common one! Thanks.

**Comment by cart** [comment]
Two small notes:

* In general I would prefer to not use macros in cases like this. It might involve a bit of extra code duplication, but it improves debuggability / legibility.
*  In this case I think the macro conversation isn't super important because this seems like the right call:

```rust

const FILE_EXTENSIONS: &[&str] = &["png", "dds", "tga", "jpg", "jpeg"];

impl AssetLoader for ImageTextureLoader {
    fn load<'a>(
        &'a self,
        bytes: &'a [u8],
        load_context: &'a mut LoadContext,
    ) -> BoxedFuture<'a, Result<()>> {
        Box::pin(async move {
            use bevy_core::AsBytes;

            // Find the image type we expect. A file with the extension "png" should
            // probably load as a PNG.

            let ext = load_context.path().extension().unwrap().to_str().unwrap();

            let img_format = image::ImageFormat::from_extension(ext)
                .ok_or_else(|| {
                    format!(
                    "Unexpected image format {:?} for file {}, this is an error in `bevy_render`.",
                    ext,
                    load_context.path().display()
                )
                })
                .unwrap();
```

**Comment by blunted2night** [comment]
That is much better, I updated the pull request.

**Pull request merged**